listener/controller/apibinding_controller_test.go

```diff
@@ -0,0 +1,76 @@
+package controller_test
+
+import (
+	"context"
+	kcpcore "github.com/kcp-dev/kcp/sdk/apis/core/v1alpha1"
+	"github.com/openmfp/kubernetes-graphql-gateway/listener/controller"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+
+	controllerRuntimeMocks "github.com/openmfp/kubernetes-graphql-gateway/gateway/resolver/mocks"
+	apischemaMocks "github.com/openmfp/kubernetes-graphql-gateway/listener/apischema/mocks"
+	clusterpathMocks "github.com/openmfp/kubernetes-graphql-gateway/listener/clusterpath/mocks"
+	discoveryclientMocks "github.com/openmfp/kubernetes-graphql-gateway/listener/discoveryclient/mocks"
+	workspacefileMocks "github.com/openmfp/kubernetes-graphql-gateway/listener/workspacefile/mocks"
+)
+
+func TestAPIBindingReconciler_Reconcile(t *testing.T) {
+	tests := []struct {
+		name        string
+		clusterName string
+		mockSetup   func(
+			io *workspacefileMocks.MockIOHandler,
+			df *discoveryclientMocks.MockFactory,
+			sc *apischemaMocks.MockResolver,
+			pr *clusterpathMocks.MockResolver,
+		)
+		err error
+	}{
+		{
+			name:        "workspace_is_deleted_ERROR",
```

Comment:
I decided to start with the testcase which checks if Delete func is called in case of deleted worksapce.
Later we can extend it to cover other cases.